### PR TITLE
TRUNK-6165 - ConceptService getConceptsByMapping should only return d…

### PR DESCRIPTION
**Pull request by mseaton**
…istinct concepts

### CODE REVIEW

File: api/src/main/java/org/openmrs/api/db/hibernate/HibernateConceptDAO.java [hunk 1]
```diff
@@ -2085,9 +2088,6 @@ private Criteria createSearchConceptMapCriteria(String code, String sourceName,
 			criteria.addOrder(Order.asc("concept.retired"));
 		}
 
-		// we only want distinct concepts
-		criteria.setResultTransformer(DistinctRootEntityResultTransformer.INSTANCE);
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/api/db/hibernate/HibernateConceptDAO.java:2089]: Did you intentionally remove this?

File: api/src/main/java/org/openmrs/api/db/hibernate/HibernateConceptDAO.java [hunk 2]
```diff
@@ -1027,7 +1028,8 @@ public void remove() {
 	public List<Concept> getConceptsByMapping(String code, String sourceName, boolean includeRetired) {
 		Criteria criteria = createSearchConceptMapCriteria(code, sourceName, includeRetired);
 		criteria.setProjection(Projections.property("concept"));
-		return (List<Concept>) criteria.list();
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/api/db/hibernate/HibernateConceptDAO.java:1030]: Did you intentionally leave out a test for this?

File: api/src/main/java/org/openmrs/api/db/hibernate/HibernateConceptDAO.java (hunk 1 above)
Comment (mseaton) [reply api/src/main/java/org/openmrs/api/db/hibernate/HibernateConceptDAO.java:2089]: Yes, but interested in your thoughts.  I think this was a holdover from an earlier time.  The root entity here is a ConceptMap, not a Concept.  I think the assumption was that this was leading the method to return distinct Concepts, but this was not actually the case.  This just resulted in distinct ConceptMaps...

File: api/src/main/java/org/openmrs/api/db/hibernate/HibernateConceptDAO.java (hunk 2 above)
Comment (mseaton) [reply api/src/main/java/org/openmrs/api/db/hibernate/HibernateConceptDAO.java:1030]: Well, I didn't intentionally add a test.  There was no existing test, and the method is deprecated, so I didn't really worry about it.

File: api/src/main/java/org/openmrs/api/db/hibernate/HibernateConceptDAO.java (hunk 1 above)
Comment (dkayiwa) [reply api/src/main/java/org/openmrs/api/db/hibernate/HibernateConceptDAO.java:2089]: I would be cautious about removing this and do so only when am very sure that it does not have unintended side effects. At the very least, i would remove it from a commit different for that of fixing duplicate concept ids.

Comment (mseaton) [reply api/src/main/java/org/openmrs/api/db/hibernate/HibernateConceptDAO.java]: OK, I added it back in @dkayiwa .  I don't think it has any impact, so it's not really a problem (though I think it is likely misleading, assuming I'm understanding it correctly).  Unit test I added passes either way...

Comment (dkayiwa) [reply api/src/main/java/org/openmrs/api/db/hibernate/HibernateConceptDAO.java]: Am fine removing it from a separate commit.

**Pull request merged**